src/main/kotlin/nl/avisi/structurizr/site/generatr/StructurizrUtilities.kt

```diff
@@ -36,14 +38,43 @@ fun Container.hasSections() = documentation.sections.isNotEmpty()
 fun ViewSet.hasSystemContextViews(softwareSystem: SoftwareSystem) =
     systemContextViews.any { it.softwareSystem == softwareSystem }
 
-fun ViewSet.hasContainerViews(softwareSystem: SoftwareSystem) =
-    containerViews.any { it.softwareSystem == softwareSystem }
+fun ViewSet.hasContainerViews(generatorContext: GeneratorContext, softwareSystem: SoftwareSystem) =
+    containerViews.any { it.softwareSystem == softwareSystem } || getImagesForSystem(generatorContext, softwareSystem).isNotEmpty()
 
-fun ViewSet.hasComponentViews(softwareSystem: SoftwareSystem) =
-    componentViews.any { it.softwareSystem == softwareSystem }
+fun ViewSet.hasComponentViews(generatorContext: GeneratorContext, softwareSystem: SoftwareSystem) =
+    componentViews.any { it.softwareSystem == softwareSystem } || getImagesForContainer(generatorContext, softwareSystem).isNotEmpty()
+
+fun ViewSet.hasCodeViews(generatorContext: GeneratorContext, softwareSystem: SoftwareSystem) =
+    componentViews.any { it.softwareSystem == softwareSystem } || getImagesForComponent(generatorContext, softwareSystem).isNotEmpty()
 
 fun ViewSet.hasDynamicViews(softwareSystem: SoftwareSystem) =
     dynamicViews.any { it.softwareSystem == softwareSystem }
 
 fun ViewSet.hasDeploymentViews(softwareSystem: SoftwareSystem) =
     deploymentViews.any { it.softwareSystem == softwareSystem }
+
+fun getImagesForSystem(generatorContext: GeneratorContext, softwareSystem: SoftwareSystem): List<ImageView> {
+    return generatorContext.workspace.views.imageViews
+        .filter { it.elementId == softwareSystem.id }
+        .sortedBy { it.key }
+}
+
+fun getImagesForContainer(generatorContext: GeneratorContext, softwareSystem: SoftwareSystem): List<ImageView> {
```

Comment:
@jp7677 Is there a better way to write this in Kotlin?

Reply:
From a short glimpse, Kotlin's `flatMap` should be your friend here https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/flat-map.html